gateway-messages/tests/versioning/v19.rs

```diff
@@ -0,0 +1,108 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! This source file is named after the protocol version being tested,
+//! e.g. v01.rs implements tests for protocol version 1.
+//! The tested protocol version is represented by "$VERSION" below.
+//!
+//! The tests in this module check that the serialized form of messages from MGS
+//! protocol version $VERSION have not changed.
+//!
+//! If a test in this module fails, _do not change the test_! This means you
+//! have changed, deleted, or reordered an existing message type or enum
+//! variant, and you should revert that change. This will remain true until we
+//! bump the `version::MIN` to a value higher than $VERSION, at which point these
+//! tests can be removed as we will stop supporting $VERSION.
+
+use super::assert_serialized;
+use gateway_messages::Fwid;
+use gateway_messages::ImageError;
+use gateway_messages::MgsRequest;
+use gateway_messages::RotBootInfo;
+use gateway_messages::RotSlotId;
+use gateway_messages::RotStateV3;
+use gateway_messages::SpResponse;
+use gateway_messages::SpStateV3;
+use gateway_messages::UpdateError;
+
+#[test]
```

Comment:
Versioning tests should be limited to testing what changed in this specific version; we shouldn't be testing `SpResponse`, `MgsRequest`, or `ImageError`.

In `v19.rs`, we should only test `UpdateError::InvalidPreferredSlotId`.  While looking at this, I noticed that there are untested variants, so I opened #400 to fix them.